### PR TITLE
Case 22334: openUrl fix

### PR DESCRIPTION
**Pull request by SamGondelman**
https://highfidelity.fogbugz.com/f/cases/22334

- Removes LoginDialog::openUrl and Application::openUrl
- FileDialogHelper::openDirectory only works on directories
- Sets up a URL handler for links with the following schemes: file, all of your drives, empty schemes

Test plan:
- `Window.openUrl("hifi://eschatology")` should bring you to eschatology
- `Window.openUrl("hifiapp:GOTO")` should open GOTO
- `Window.openUrl("http://google.com")` and `Window.openUrl("https://google.com")` should open google in your default browser
- `Window.openUrl("mailto:support@highfidelity.com")` will open your mail app or whatever you have configured to handle mailto links
- The following will open a dialog asking you to confirm your choice.  If you select "no", nothing will happen.  If you select "yes", that directory will be opened (or nothing will happen if the directory doesn't exist)
	- `Window.openUrl("file:///c:/users")`
	- `Window.openUrl("c:/Users")`
	- `Window.openUrl("asdf")`
- Download https://hifi-content.s3.amazonaws.com/samuel/openUrlExternally.qml and run this from the console (don't just run it from the url!):
```
var window = new OverlayWindow({
    title: "test",
    source: "file:///<link to the qml>",
    width: 300,
    height: 50
});
```
- You'll see a window with 6 buttons.  The first two will open google, the third will open your mail app, and the last three should bring up confirmation dialog boxes (two of which will open a directory in File Explorer, the last of which won't do anything)
- In the Running Scripts window, at the bottom, "Reveal Scripts Folder" will open the scripts folder, without a confirmation dialog.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18977/HighFidelity-Beta-PR15456-31b3f0e.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18977/HighFidelity-Beta-Interface-PR15456-31b3f0e.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18977/HighFidelity-Beta-PR15456-31b3f0e.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18977/HighFidelity-Beta-Interface-PR15456-31b3f0e.exe

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -2443,6 +2443,13 @@ Application::Application(int& argc, char** argv, QElapsedTimer& startupTimer, bo
     DependencyManager::get<TabletScriptingInterface>()->preloadSounds();
     DependencyManager::get<Keyboard>()->createKeyboard();
 
+    QDesktopServices::setUrlHandler("file", this, "showUrlHandler");
+    QDesktopServices::setUrlHandler("", this, "showUrlHandler");
+    auto drives = QDir::drives();
+    for (auto drive : drives) {
+        QDesktopServices::setUrlHandler(QUrl(drive.absolutePath()).scheme(), this, "showUrlHandler");
+    }
```
Comment (SamGondelman) [inline interface/src/Application.cpp]: Local file can take the following forms:
- `file://something`
- `c:/something` etc.
- `something` (treated as relative)

**Comment by SamGondelman** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18981/HighFidelity-Beta-PR15456-31b3f0e.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18981/HighFidelity-Beta-Interface-PR15456-31b3f0e.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18981/HighFidelity-Beta-PR15456-31b3f0e.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18981/HighFidelity-Quest-Beta-PR15456-31b3f0e.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18981/HighFidelity-Beta-PR15456-31b3f0e.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18981/HighFidelity-Beta-Interface-PR15456-31b3f0e.exe

File: interface/src/Application.cpp (hunk 1 above)
Comment (NissimHadar) [reply interface/src/Application.cpp]: Isn't it file:///?

Comment (SamGondelman) [reply interface/src/Application.cpp]: Yes, updated my comment.  The only place in the code that depends on that is Application::openDiretory, which I just fixed to use 3 slashes

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18988/HighFidelity-Beta-PR15456-8253d2f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18988/HighFidelity-Beta-Interface-PR15456-8253d2f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18988/HighFidelity-Beta-PR15456-8253d2f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18988/HighFidelity-Beta-Interface-PR15456-8253d2f.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18988/HighFidelity-Beta-PR15456-8253d2f.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18988/HighFidelity-Quest-Beta-PR15456-8253d2f.apk)

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18991/HighFidelity-Beta-PR15456-ef64de7.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18991/HighFidelity-Quest-Beta-PR15456-ef64de7.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18991/HighFidelity-Beta-PR15456-ef64de7.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18991/HighFidelity-Beta-Interface-PR15456-ef64de7.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18991/HighFidelity-Beta-PR15456-ef64de7.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18991/HighFidelity-Beta-Interface-PR15456-ef64de7.exe

**Pull request merged**